src/alephbet.coffee

```diff
@@ -59,25 +61,30 @@ class AlephBet
       return unless variant
       @storage().set("#{@options.name}:#{goal_name}", true) if props.unique
       utils.log("experiment: #{@options.name} variant:#{variant} goal:#{goal_name} complete")
-      @tracking().goal_complete(@options.name, variant, goal_name)
+      @tracking().goal_complete(this, variant, goal_name, props)
 
     get_stored_variant: ->
       @storage().get("#{@options.name}:variant")
 
     pick_variant: ->
       partitions = 1.0 / @variant_names.length
-      chosen_partition = Math.floor(Math.random() / partitions)
+      chosen_partition = Math.floor(@_random('variant') / partitions)
       variant = @variant_names[chosen_partition]
       utils.log("#{variant} picked")
       variant
 
     in_sample: ->
       active = @storage().get("#{@options.name}:in_sample")
       return active unless typeof active is 'undefined'
-      active = Math.random() <= @options.sample
+      active = @_random('sample') <= @options.sample
       @storage().set("#{@options.name}:in_sample", active)
       active
 
+    _random: (salt) ->
+      return utils.random() unless @user_id
+      seed = "#{@name}.#{salt}.#{@user_id}"
+      utils.random(seed)
```

Comment:
Are you asking why I'm assigning it to a variable? not strictly necessary. I can pass it directly as you suggested. But might be a bit more readable to understand what this string is used for?